lib/Ohmbrewer_Thermostat.cpp

```diff
@@ -292,24 +322,24 @@ int Ohmbrewer::Thermostat::doWork() {
     //TURN ON
     if (getState() && gap!=0) {//if we want to turn on the element (thermostat is ON)
         //TURN ON state and powerPin
-        if (!(_heatingElm->getState())) {//if heating element is off
-            _heatingElm->setState(true);//turn it on
-            if (_heatingElm->getPowerPin() != -1) { // if powerPin enabled
-                digitalWrite(_heatingElm->getPowerPin(), HIGH); //turn it on (only once each time you switch state)
+        if (!(getElement()->getState())) {//if heating element is off
+            getElement()->setState(true);//turn it on
+            if (getElement()->getPowerPin() != -1) { // if powerPin enabled
+                digitalWrite(getElement()->getPowerPin(), HIGH); //turn it on (only once each time you switch state)
             }
         }
         //RELAY MODULATION
         if (output < millis() - windowStartTime) {
-            digitalWrite(_heatingElm->getControlPin(), HIGH);
+            digitalWrite(getElement()->getControlPin(), HIGH);
         } else {
-            digitalWrite(_heatingElm->getControlPin(), LOW);
+            digitalWrite(getElement()->getControlPin(), LOW);
         }
     }
     //TURN OFF
-    if (gap == 0) {//once reached target temp
-        _heatingElm->setState(false); //turn off element
-        if (_heatingElm->getPowerPin() != -1) { // if powerPin enabled
-            digitalWrite(_heatingElm->getPowerPin(), LOW); //turn it off too
+    if (gap == 0 || getTargetTemp()->c() <= getSensor()->getTemp()->c() ) {//once reached target temp
+        getElement()->setState(false); //turn off element
+        if (getElement()->getPowerPin() != -1) { // if powerPin enabled
+            digitalWrite(getElement()->getPowerPin(), LOW); //turn it off too
```

Comment:
I don't think you want to mess with the digitalWrite for the power pin here. The Element's doWork function should handle it's pins appropriately depending on its given state.